### PR TITLE
Extend EDB data and correct multiphase EDB handling

### LINKED ISSUE #277: Misc EDB corrections
These are a series of issues that came up while testing EDB for common usage cases. All will be resolved in (#274).

- [x] Update 'get_reactions' function to disregard reactions that involve an invalid/unwanted phase
- [x] Update 'get_reactions' only iterate on items and not items and phases (this behavior is redundant) 
- [x] Update 'schema' to reflect changes in how phase equilibrium is handled 
- [x] Update 'data_model' to correctly form tuple keys for phase equilibrium in components and base configs 
- [x] Add new IDAES objects to the set of replacements in 'data_model'
- [x] Update edb unit tests to reflect schema changes
- [x] Create new unit tests for edb examples
- [x] Add documentation/juypter notebooks for examples

### PR DESCRIPTION
**Pull request by aladshaw3**
## Fixes/Addresses:

Extends the EDB default database to include all components and reactions of interest from BP1. Adds corrections to EDB behavior for handling multiple phases. 

## Summary/Motivation:

Now that we will be using the EDB more, we should have our default database better reflect the chemistry that we have tested with from BP1. This would cover the most important chemical species of concern from BP1. More may be periodically added in BP2 as we add new chemical models 

## Changes proposed in this PR:
- Added solid species to database
- Added iron species to database
- Added solubility reactions to database 
- Added iron and boron speciation reactions to database
- Updated schema to allow for inclusion of solids 
- Added unit tests for EDB to build valid configurations of Vap-Liq and Sol-Liq systems
- Added unit tests differentiating between inherent and equilibrium reactions 
- Changed the schema for Vap-Liq handling 
- Updated data_model to appropriately build Vap-Liq equilibrium configs
- Updated get_reactions function to properly handle list of valid phases
- Adding new Juypter notebooks to demo EDB

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/edb/db_api.py [hunk 1]
```diff
@@ -276,34 +275,43 @@ def get_reactions(
...
-                    stoich = item[stoich_field][phase]
-                    if any_components:
-                        # look for non-empty intersection
-                        if set(stoich.keys()) & cnames:
-                            found.append(item)
+                        disallow = True
+                    for n in item[stoich_field][phase]:
+                        stoich[n] = item[stoich_field][phase][n]
+                #If the item involves a phase that is not allowed, then move on to next item
+                if (disallow):
```
Comment (dangunter) [inline watertap/edb/db_api.py:286]: minor syntax point: parens are not needed here

File: docs/examples/edb/edb_example.ipynb [hunk 2]
```diff
@@ -463,7 +463,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.6"
+   "version": "3.8.12"
```
Comment (dangunter) [inline docs/examples/edb/edb_example.ipynb:466]: this is fine, but as an aside I wish we didn't have this sort of thing show up as a "change"

**Review by dangunter — APPROVED** [review]
This looks great

File: watertap/edb/db_api.py (hunk 1 above)
Comment (aladshaw3) [reply watertap/edb/db_api.py:286]: Noted. I can clean this up later. That way it doesn't automatically disregard your current review.

File: docs/examples/edb/edb_example.ipynb (hunk 2 above)
Comment (aladshaw3) [reply docs/examples/edb/edb_example.ipynb:466]: Noted.

Comment (srikanthallu) [inline docs/examples/edb/edb_example.ipynb:466]: Was there a reason to revert back.?

**Review by srikanthallu — APPROVED** [review]
LGTM

File: docs/examples/edb/edb_example.ipynb (hunk 2 above)
Comment (aladshaw3) [reply docs/examples/edb/edb_example.ipynb:466]: This is apparently just metadata that gets added to the notebook depending on who last opened/edited it. This file was originally created by Dan, but when I edited it it got reverted to my version.

**Pull request merged**